### PR TITLE
[RPC] Add Streaming RPC Types

### PR DESCRIPTION
**Pull request by CapCap**
## Motivation
Refactor the streaming RPC types so we can expose them (there is a cyclical dep with the `client` library otherwise)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan
Existing tests

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Changing JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M diem-json-rpc-types 0.0.3 (workspace, path 'json-rpc/types')
    * version upgraded from 0.0.2
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M diem-json-rpc-types 0.0.3 (workspace, path 'json-rpc/types')
    * version upgraded from 0.0.2
    * (unchanged features: [none])
```

**Review by xli — APPROVED** [review]

File: json-rpc/types/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "diem-json-rpc-types"
-version = "0.0.2"
+version = "0.0.3"
```
Comment (bmwill) [inline json-rpc/types/Cargo.toml:3]: can this be changed back? We never bump this until a release is made.
@CapCap

**Pull request closed without merging**